Rewrite Python CGI under http/tests/multipart for Windows Python

fujii opened the pull request:
#### 903c53aa4abcd48c2108c02ab063043c746cd91c
<pre>
Rewrite Python CGI under http/tests/multipart for Windows Python
<a href="https://bugs.webkit.org/show_bug.cgi?id=263425">https://bugs.webkit.org/show_bug.cgi?id=263425</a>

Reviewed by Ross Kirsling.

Windows Python replaces &apos;\n&apos; to &apos;\r\n&apos; for sys.stdout. Use
sys.stdout.buffer.write instead of sys.stdout.write.

* LayoutTests/http/tests/multipart/load-last-non-html-frame.py:
* LayoutTests/http/tests/multipart/multipart-html.py:
* LayoutTests/http/tests/multipart/multipart-replace-non-html-content.py:
* LayoutTests/http/tests/multipart/policy-ignore-crash.py:
* LayoutTests/http/tests/multipart/resources/multipart-nodashes.py:
* LayoutTests/http/tests/multipart/resources/multipart-wait-before-boundary.py:
* LayoutTests/http/tests/multipart/resources/multipart.py:
* LayoutTests/platform/wincairo/TestExpectations:
* LayoutTests/platform/wincairo/http/tests/multipart/invalid-image-data-standalone-expected.txt:

Canonical link: <a href="https://commits.webkit.org/269589@main">https://commits.webkit.org/269589@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a2ef68e61ebfb24b9afb4ed34a1a3200ef2c02e3

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/22903 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/916 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/23999 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/24812 "Built successfully") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/21202 "Build is in progress. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request; Running compile-webkit") 
| | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/49/builds/2318 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/23437 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/22096 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/23143 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/453 "Found 39 new test failures: accessibility/ios-simulator/inline-prediction-attributed-string.html, editing/caret/ios/absolute-caret-position-after-scroll.html, editing/caret/ios/caret-color-after-refocusing-input.html, editing/caret/ios/caret-color-auto.html, editing/caret/ios/caret-color-currentcolor.html, editing/caret/ios/caret-color-in-nested-editable-containers.html, editing/caret/ios/caret-in-overflow-area.html, editing/selection/character-granularity-rect.html, editing/selection/ios/absolute-selection-after-scroll.html, editing/selection/ios/become-first-responder-after-relinquishing-focus.html ... (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/19868 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/25670 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/358 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/20746 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/26946 "Passed tests") | 
| | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/20744 "Build is in progress. Recent messages:OS: Sonoma (14.0), Xcode: 15.0; Skipping applying patch since patch_id isn't provided; Checked out pull request; Running run-api-tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/21004 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/24802 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/437 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/18251 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/354 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/848 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/2913 "Built successfully and passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/605 "Build is in progress. Recent messages:OS: Sonoma (14.0), Xcode: 15.0; Skipping applying patch since patch_id isn't provided; Checked out pull request; Running compile-webkit") | | | 
<!--EWS-Status-Bubble-End-->

rkirsling reviewed (approved):
This seems fine from my perspective; is this something that Tools/Scripts/webkitpy has also needed to worry about previously?

fujii committed:
Rewrite Python CGI under http/tests/multipart for Windows Python
https://bugs.webkit.org/show_bug.cgi?id=263425

Reviewed by Ross Kirsling.

Windows Python replaces '\n' to '\r\n' for sys.stdout. Use
sys.stdout.buffer.write instead of sys.stdout.write.

* LayoutTests/http/tests/multipart/load-last-non-html-frame.py:
* LayoutTests/http/tests/multipart/multipart-html.py:
* LayoutTests/http/tests/multipart/multipart-replace-non-html-content.py:
* LayoutTests/http/tests/multipart/policy-ignore-crash.py:
* LayoutTests/http/tests/multipart/resources/multipart-nodashes.py:
* LayoutTests/http/tests/multipart/resources/multipart-wait-before-boundary.py:
* LayoutTests/http/tests/multipart/resources/multipart.py:
* LayoutTests/platform/wincairo/TestExpectations:
* LayoutTests/platform/wincairo/http/tests/multipart/invalid-image-data-standalone-expected.txt:

Canonical link: https://commits.webkit.org/269589@main
